### DISCUSSION TITLE
Prefixes, Suffixes & Groups

### DISCUSSION THREAD

**Discussion by ollie7671** [Axieum/Chatter#38] [Q&A]
Hi,
I know there is a short description of how to setup custom in game chat messages but I was wondering if you could show how to do prefixes? I tried what I thought I was supposed to do but I couldn't figure it out.

Thank you!

**Comment by axieum** [comment]
Hey, 👋

I'd recommend installing the [LuckPerms](https://luckperms.net) mod on your server and following their prefix, suffix and groups tutorial.

Chatter will automatically detect LuckPerms and start providing prefixes, etc.

Hope this helps,
Ax.

**Reply by axieum** [reply]
```jsonc
/**
 * The in-game chat message JSON template (see https://minecraftjson.com)
 * Use ${username}, ${player}, ${message}, ${group}, ${prefix}, ${suffix}, ${team}, ${team_color}, ${world} and ${datetime[:format]}
 **/
"template": "[\"\",{\"text\":\"[${group}] \"},{\"text\":\"${prefix}\"},{\"text\":\"${player}\",\"color\":\"yellow\",\"clickEvent\":{\"action\":\"suggest_command\",\"value\":\"/tell ${username} \"},\"hoverEvent\":{\"action\":\"show_text\",\"contents\":[\"\",{\"text\":\"Click to direct message\",\"italic\":true}]}},{\"text\":\"${suffix}\"},{\"text\":\" > \",\"color\":\"dark_gray\"},{\"text\":\"${message}\"}]",
```

**Reply by ollie7671** [reply]
Thanks, I will give it a go!!

**Reply by ollie7671** [reply]
Hi again!! I have tried your solution and I haven't gotten it to work. Is there any chance I can add you on discord or anything so that we could further talk about it?

**Reply by axieum** [reply]
Sure, `@Axieum#1001` 😊

**Reply by ollie7671** [reply]
Awesome thank you I added you. I am FamousManatee :)
